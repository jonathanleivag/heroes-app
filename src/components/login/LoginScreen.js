import React from 'react'

export const LoginScreen = ({ history: { replace } }) => {
    return (
        <section className='container mt-5'>
            <h1>Login</h1>
            <hr />
            <button
                className='btn btn-outline-success'
                onClick={() => replace('/')}
            >
                Login
            </button>
        </section>
    )
}
