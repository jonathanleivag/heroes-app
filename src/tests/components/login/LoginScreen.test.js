import '@testing-library/jest-dom'
import React from 'react';
import { mount } from 'enzyme'
import { LoginScreen } from '../../../components/login/LoginScreen';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';

describe('Pruebas en <LoginScreen />', () => {

    const history = {
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={history} />
        </AuthContext.Provider>
    )

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe de realizar el dispatch y la navegaciÃ³n', () => {
       const handlerClick = wrapper.find('button').prop('onClick')
       handlerClick()
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: { name: 'Jonathan' }
        })

        localStorage.setItem('lastPath','/marvel')
        expect(history.replace).toHaveBeenCalledWith('/')
        handlerClick()
        expect(history.replace).toHaveBeenCalledWith('/marvel')

    })

})
