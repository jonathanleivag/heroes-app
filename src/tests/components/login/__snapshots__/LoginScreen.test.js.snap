// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe de mostrarse correctamente 1`] = `
<section
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-outline-success"
    onClick={[Function]}
  >
    Login
  </button>
</section>
`;
